zilliqa/src/consensus.rs

```diff
@@ -1056,11 +1109,61 @@ impl Consensus {
         // need those transactions again
         for tx in opaque_transactions {
             let account_nonce = self.state.get_account(tx.signer)?.nonce;
-            self.transaction_pool.insert_transaction(tx, account_nonce);
+            transaction_pool.insert_transaction(tx, account_nonce);
         }
         Ok(Some((proposal, broadcasted_transactions)))
     }
 
+    fn propose_new_block(&mut self) -> Result<Option<(Block, Vec<VerifiedTransaction>)>> {
+        let mut state = self.state.clone();
+        let mut tx_pool = self.transaction_pool.clone();
+        let mut votes = self.votes.clone();
+
+        let result = self.propose_new_block_at(&mut state, &mut tx_pool, &mut votes);
+
+        self.state = state;
+        self.transaction_pool = tx_pool;
+        self.votes = votes;
+
+        result
+    }
+
+    pub fn propose_pending_block(&self) -> Result<Option<Block>> {
```

Comment:
Also very minor nit but maybe this could be named something like `get_pending_block()`? It doesn't even return a proper `Proposal` since it doesn't return the transactions. And it's a bit odd seeing the eth API make a call to a method called "propose a block"